crates/natrix/tests/actual_tests/derive_macros.rs

```diff
@@ -0,0 +1,221 @@
+use natrix::prelude::*;
+use wasm_bindgen_test::{wasm_bindgen_test, wasm_bindgen_test_configure};
+
+wasm_bindgen_test_configure!(run_in_browser);
+
+// Test the Project derive macro with additional derives
+#[derive(Project)]
+#[project(Debug)]
+enum TestEnum<T> {
+    Value(T),
+    Text(String),
+    Empty,
+}
+
+#[derive(Project)]
+enum SimpleEnum {
+    A,
+    B,
+    C,
+}
+
+// Test Downgrade derive
+#[derive(Downgrade)]
+enum DowngradeTestEnum<T> {
+    Value(T),
+    Empty,
+}
```

Comment:
I know the "stdlib" downgrade impls are this generic, but the intended use of the derive macro is on the enum generated by `Project`, i.e a enum with `Ref` fields.